libraries/AP_LQR_Control/AP_LQR_Control.cpp

```diff
@@ -0,0 +1,409 @@
+// Bank angle command based on adaptive optimal control.
+// Akshath Singhal 2018
+
+#include <AP_HAL/AP_HAL.h>
+#include "AP_LQR_Control.h"
+
+extern const AP_HAL::HAL& hal;
+
+// table of user settable parameters
+const AP_Param::GroupInfo AP_LQR_Control::var_info[] = {
+    
+    // @Param: LIM_BANK
+    // @DisplayName: Loiter Radius Bank Angle Limit
+    // @Description: The sealevel bank angle limit for a continous loiter. (Used to calculate airframe loading limits at higher altitudes). Setting to 0, will instead just scale the loiter radius directly
+    // @Units: deg
+    // @Range: 0 89
+    // @User: Advanced
+    AP_GROUPINFO_FRAME("LIM_BANK",   1, AP_LQR_Control, _loiter_bank_limit, 0.0f, AP_PARAM_FRAME_PLANE),
+
+    // @Param: LQR_MAX_XTRACK
+    // @DisplayName: LQR maximum allowed Crosstrack error
+    // @Description: Maximum allowed crosstrack error(in centimeters) for LQR. A very small value can lead to high frequency of oscillations. A very large value leads to slow convergence to expected path.
+    // @Range: 400 10000
+    // @Units: cm
+    // @Increment: 20
+    // @User: Advanced
+    AP_GROUPINFO("MAX_XTRCK",   2, AP_LQR_Control, _max_xtrack, 2000),
+
+    // @Param: LQR_Q2_VAL
+    // @DisplayName: LQR VALUE of Q2
+    // @Description: The value of Q2 to be used. Divided by 100 before use.
+    // @Range: 1 10000
+    // @Increment: 10
+    // @User: Advanced
+    AP_GROUPINFO("Q2_VAL",   3, AP_LQR_Control, _q2_val, 100),
+
+    // @Param: LQR_XTRACK_FACTOR
+    // @DisplayName: LQR XTRACK Error weightage in acceleration.
+    // @Description: The affect of Xtrackk error on lateral acceleration. Default is 100. Divided by 100 before use.
+    // @Range: 10 1000
+    // @Increment: 5
+    // @User: Advanced
+    AP_GROUPINFO("XTR_FAC",   4, AP_LQR_Control, _xtrack_fac, 100),
+
+    // @Param: LQR_VELOCITY_FACTOR
+    // @DisplayName: Weightage of rate of change of Xtrack in acceleration.
+    // @Description: The affect of rate of change of Xtrackk error on lateral acceleration. Default is 150. Divided by 100 before use.
+    // @Range: 10 1000
+    // @Increment: 5
+    // @User: Advanced
+    AP_GROUPINFO("VEL_FAC",   5, AP_LQR_Control, _vel_fac, 150),
+
+    // @Param: LQR_MAX_LATACC
+    // @DisplayName: LQR maximum allowed lateral acceleration
+    // @Description: Maximum allowed lateral acceleration(in centimeters/sec sqaure) for LQR. A very small value can lead to slow turning and higher overshoots. A very large value leads to very small turning radius causing unstable roll and yaw throws.
```

Comment:
remove the redundant units in the description

Reply:
Done